exe/3scale-update

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env ruby
+
+require '3scale_toolbox/cli'
+require 'optparse'
+require '3scale/api'
+require 'uri'
+
+options = {}
+
+parser = OptionParser.new do |parser|
+  parser.banner = '3scale update <command> [options]'
+
+  parser.on('-s', '--source SOURCE', "Source") do |domain|
+    options[:source] = domain
+  end
+
+  parser.on('-d', '--destination DESTINATION', "Destination") do |domain|
+    options[:destination] = domain
+  end
+
+  parser.on('-f', '--force', 'Overwrites the mapping rules by deleting all rules from target service first') do
+    options[:force] = true
+  end
+
+  # TODO: parametrize what parts of service need to be copied 
+  parser.on('-r', '--rules-only', 'Updates only the mapping rules') do
+    options[:rules_only] = true
+  end
+
+  parser.on('-h', '--help', 'Prints this help') do
+    puts parser
+    puts
+    puts 'Available Commands:', ['service <source_service_id> <destination_service_id>', 'help']
+    exit
+  end
+end
+
+print_help = ->(error = nil) do
+  if error
+    puts "Error: #{error}"
+    puts
+  end
+  parser.parse(['--help'])
+end
+
+parser.parse!
+
+def fetch_option(options, key)
+  options.fetch(key) { raise OptionParser::MissingArgument, key }
+end
+
+class ServiceUpdater
+
+  attr_reader :source_client, :target_client, :source_service_id, :target_service_id
+
+  def initialize (source, source_service_id, destination, target_service_id)
+    @source_client = ThreeScale::API.new(
+      endpoint:     endpoint_from_url(source), 
+      provider_key: provider_key_from_url(source)
+    )
+    @target_client = ThreeScale::API.new(
+      endpoint:     endpoint_from_url(destination), 
+      provider_key: provider_key_from_url(destination)
+    )
+    @source_service_id = source_service_id
+    @target_service_id = target_service_id
+  end
+
+  def compare_hashes(first, second, keys)
+    keys.map{ |key| first.fetch(key) } == keys.map{ |key| second.fetch(key) }
+  end
+
+  def provider_key_from_url(url)
+    URI(url).user
+  end
+
+  def endpoint_from_url(url)
+    uri      = URI(url)
+    uri.user = nil
+    
+    uri.to_s
+  end
+
+  def target_service_params(source)
+    # NOTE: backend_version and deployment_option are not yet returned by show_service method
+    params = %w(name backend_version deployment_option end_user_registration_required)
+    Hash[params.collect {|p| [p, source[p]]}].reject { |key, value| !value }
```

Comment:
Might be better to use `Array#to_h` instead of `Hash[]` just for consistency with other methods.